src/python/aqueduct_executor/operators/function_executor/execute.py

```diff
@@ -66,11 +71,41 @@ def _import_invoke_method(spec: FunctionSpec) -> Callable[..., DataFrame]:
     return getattr(function, method_name)  # type: ignore
 
 
+def _infer_result_type(result: Any) -> ArtifactType:
+    if isinstance(result, DataFrame):
+        return ArtifactType.TABULAR
+    elif isinstance(result, Image.Image):
+        return ArtifactType.IMAGE
+    elif isinstance(result, bytes):
+        return ArtifactType.BYTES
+    elif isinstance(result, str):
+        # We first check if the result is a valid JSON string.
+        try:
+            json.loads(result)
+            return ArtifactType.JSON
+        except:
+            return ArtifactType.STRING
+    elif isinstance(result, bool) or isinstance(result, np.bool_):
+        return ArtifactType.BOOL
+    elif isinstance(result, int) or isinstance(result, float) or isinstance(result, np.number):
```

Comment:
These if statements are shared with the extraction code right? Can we put them in a shared helper to prevent divergence?

Reply:
I remember there's a bit of a difference in the use case, but will try to see if we can consolidate them.

Reply:
They look similar but don't serve the same goal, plus the one in the connector side is only a subset of the one in the function executor, so I think having them separated makes more sense now.